From 0acbbf616f590d59e6e726a5d98d5efed1321c8f Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Wed, 19 Aug 2020 09:43:12 +0200
Subject: [PATCH 09/31] common: replace forceUTF8Locale's use of strncpy

strncpy could leave the output buffer without a \0 if the input was
large enough.
---
 common/charset/localeutil.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/common/charset/localeutil.cpp b/common/charset/localeutil.cpp
index b16753018..ccc232c1d 100644
--- a/common/charset/localeutil.cpp
+++ b/common/charset/localeutil.cpp
@@ -8,6 +8,7 @@
 #include <clocale>
 #include <cstring>
 #include <unistd.h>
+#include <libHX/string.h>
 #include "localeutil.h"
 
 namespace KC {
@@ -59,7 +60,7 @@ bool forceUTF8Locale(bool bOutput, std::string *lpstrLastSetLocale)
 			std::cerr << "Unable to initialize locale" << std::endl;
 		return false;
 	}
-	strncpy(old_locale, orig_locale, sizeof(old_locale));
+	HX_strlcpy(old_locale, orig_locale, sizeof(old_locale));
 	char *dot = strchr(old_locale, '.');
 	if (dot) {
 		*dot = '\0';
-- 
2.29.2

